lib/rules/no-semantic-errors.js

```diff
@@ -24,12 +25,7 @@ module.exports = {
     create(context) {
         const sourceCode = context.getSourceCode();
 
-        const program =
-            context.parserServices && context.parserServices.program;
-
-        if (!program) {
-            return [];
-        }
+        const program = util.getParserServices(context).program;
```

Comment:
Would just `util.getProgram(context)` be a more ergonomic API?

Reply:
not really we need access to all 3 properties like here: https://github.com/bradzacher/eslint-plugin-typescript/pull/230/files#diff-dc0404c0ba0a0d7ea3cddcb1e786200fR34 or here https://github.com/bradzacher/eslint-plugin-typescript/pull/250/files#diff-a1b523ce3e593ac75a3d6205533c3022R43

Reply:
but we could always specify multiple helpers for this `util.getProgram(context)`, `util.getTSNodeFromESTreeNode(context, node)`, `util.getESNodeFromTSTreeNode(context, node)`

hmm...